td-agent/Rakefile

```diff
@@ -665,11 +665,28 @@ class BuildTask
       if File.exist?("#{gem_dir}/ext")
         rm_f(Dir.glob("#{gem_dir}/ext/**/*.o"))
         rm_f(Dir.glob("#{gem_dir}/ext/**/*.la"))
+        rm_f(Dir.glob("#{gem_dir}/ext/**/*.a"))
         rm_f(Dir.glob("#{gem_dir}/ext/**/.dirstamp"))
         rm_f(Dir.glob("#{gem_dir}/ext/**/.sitearchdir*.time"))
+        rm_rf(Dir.glob("#{gem_dir}/ext/**/.libs"))
+        rm_rf(Dir.glob("#{gem_dir}/ext/**/.deps"))
       end
-      rm_rf(["#{gem_dir}/test", "#{gem_dir}/spec", "#{gem_dir}/**/gem.build_complete"])
+      rm_rf(["#{gem_dir}/test", "#{gem_dir}/spec"])
       rm_rf(Dir.glob("#{gem_staging_dir}/**/\.git*"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/gem.build_complete"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.circleci"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.clang-format"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.document"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.gemtest"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.editorconfig"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.readygo"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.rspec"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.ruby-version"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.travis"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.yardopts"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/\.*.yml"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/*.h"))
+      rm_f(Dir.glob("#{gem_staging_dir}/**/*.c"))
```

Comment:
Could you check & clean up this method (`remove_needless_files`)?
Because current code performs redundant `glob`.
For example, `gem_staging_dir` shouldn't be used here since it's in the `Dir.glob("#{gem_staging_dir}/gems/*").each do |gem_dir|` block, and probably most of these files exist only under `#{gem_dir}` in this context.

Reply:
Thanks, I'll fix them

Reply:
I've rebased and push -f to fix them.

Reply:
I've committed more simplified code.